### DISCUSSION TITLE
Can I change sql mode in Planetscale?

### DISCUSSION THREAD

**Discussion by cpxdevth** [planetscale/discussion#375] [Q&A] [answered]
I found error when I try to select all records

`Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'table.field' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by`

So in many stackoverflow articles recommended to change sql mode but it temporary worked. After a moment It will be back to same error.
So Can I change sql mode or any solution to solve this?

**Comment by jgreet** [comment, marked as answer]
The global sql_mode of a PlanetScale database can't be changed permanently but you can modify the sql_mode of the current session. 

If your application requires you to remove ONLY_FULL_GROUP_BY you could run the following statement immediately after opening each connection:
 
```sql
set sql_mode = REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY','');
```
 
See also https://github.com/planetscale/discussion/discussions/304

**Reply by cpxdevth** [reply]
> The global sql_mode of a PlanetScale database can't be changed permanently but you can modify the sql_mode of the current session. 
> 
> If your application requires you to remove ONLY_FULL_GROUP_BY you could run the following statement immediately after opening each connection:
>  
> ```sql
> set sql_mode = REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY','');
> ```
>  
> See also https://github.com/planetscale/discussion/discussions/304

Hi Jgreet, Thank you for your solution. I already found that I need to set change sql_mode for current session is the best way for this. I cannot chnage it globally like you tell.
